documentation/src/main/asciidoc/topics/con_file_based_cache_stores.adoc

```diff
@@ -1,72 +1,36 @@
-====
-
-[discrete]
-== Soft-Index File Stores
-
-`SoftIndexFileStore` is the default implementation for file-based cache stores and stores data in a set of append-only files.
-
-When append-only files:
-
-* Reach their maximum size, {brandname} creates a new file and starts writing to it.
-* Reach the compaction threshold of less than 50% usage, {brandname} overwrites the entries to a new file and then deletes the old file.
-
-.B+ trees
-
-To improve performance, append-only files in a `SoftIndexFileStore` are indexed using a **B+ Tree** that can be stored both on disk and in memory.
-The in-memory index uses Java soft references to ensure it can be rebuilt if removed by Garbage Collection (GC) then requested again.
+The following types of file-based cache stores exist in {brandname}:
+* Shared, where cache entries persist after a cluster restart operation. Examples include SQL, JDBC, and so on. 
+* Non-shared, where cache entries do not persist after a cluster restart operation. Examples include RocksDB, single file, and so on.
 
-Because `SoftIndexFileStore` uses Java soft references to keep indexes in memory, it helps prevent out-of-memory exceptions.
-GC removes indexes before they consume too much memory while still falling back to disk.
+File-based cache stores should always be unique to each {brandname} instance.
+If you want to use the same persistent across a cluster, configure shared storage, such as a JDBC string-based cache store.
 
-You can configure any number of B+ trees with the `segments` attribute on the `index` element declaratively or with the `indexSegments()` method programmatically.
-By default {brandname} creates up to 16 B+ trees, which means there can be up to 16 indexes.
-Having multiple indexes prevents bottlenecks from concurrent writes to an index and reduces the number of entries that {brandname} needs to keep in memory.
-As it iterates over a soft-index file store, {brandname} reads all entries in an index at the same time.
+The following limitations exist with how a non-shared store interacts in a clustered environment: 
 
-Each entry in the B+ tree is a node.
-By default, the size of each node is limited to 4096 bytes.
-`SoftIndexFileStore` throws an exception if keys are longer after serialization occurs.
+* If you do not gracefully shutdown a cache, only the last node remains active and it must store all the cache's data. In-memory and eviction limitations might impact the amount of data a node can store. When you restart the cache, you can only load data from the last active node. 
+* If you stop a running rebalancing operation during cluster shutdown, {brandname} cannot keep a fixed number of copies of any entry in a cache. When you attempt to restart the cluster, the rebalancing operation remains active and it does not perform checks on nodes that rejoin the cluster. This can lead to corrupted data or data loss for a cache.
```

Comment:
Honestly I think this is a far too convoluted process to expect users to follow. I think it would be better to advise against using non-shared stores in clustered environments when `purge-on-startup=false`

WDYT @wburns @tristantarrant ?

Reply:
Shouldn't this be something that we can failfast during config validation?

Reply:
Thanks, @ryanemerson .

I added the following to the IMPORTANT note:

> Using a non-shared store in a clustered cache environment might lead to data loss, especially in instances where you configured the store with  `purge-on-startup=false`.

@wfink , what do you think about the "if" statements. Remove them or reduce them in size? You requested an entry for a graceful shutdown scenario.

Reply:
> Shouldn't this be something that we can failfast during config validation?

Thanks for the reply, @tristantarrant . Is this a comment directed at @ryanemerson or myself. Is this a doc update request (:memo:) or technical question around a component's capabilities (:grimacing:)?

Reply:
> Shouldn't this be something that we can failfast during config validation?

@tristantarrant IMO yes, WDYT @wburns ?

Reply:
@dfitzmau I think it was aimed at me, PRs welcome though :stuck_out_tongue_winking_eye:

Reply:
Hi @wfink and @ryanemerson . Please let me know if any more changes are required on this PR. Close to approval?

Reply:
@wburns Thoughts?

Reply:
additional to ryans comment not using non-shared with purge=false I would say only use with "shutdown cluster"